JavaScript/tests/syntax_test_tsx.tsx

```diff
@@ -203,12 +203,24 @@ if (a < b || c <= d) {}
 //     ^^^^^^^ meta.tag.attributes entity.other.attribute-name
 //            ^ punctuation.definition.tag.end
 
-    <T extends {}>() => {}; // </T>;
-//  ^^^^^^^^^^^^^^^^^^^^^^ meta.function
-//  ^^^^^^^^^^^^^^ meta.function meta.generic
+    <T extends "s">() => {x}; // </T>;
+//  ^^^^^^^^^^^^^^^^^^^^^^^^ meta.function
+//  ^^^^^^^^^^^^^^^ meta.function meta.generic
 //   ^ variable.parameter.generic
 //     ^^^^^^^ storage.modifier.extends
-//             ^^ meta.function meta.generic meta.mapping
+//             ^^^ meta.function meta.generic
+
+    <T extends="s">() => {x}; // </T>;
```

Comment:
This line looks broken. Neither `()` nor `=>` have a scope assigned.

![grafik](https://user-images.githubusercontent.com/16542113/161788172-3284c811-963e-4585-84c1-e05463532136.png)

Reply:
It is expected. The entirety of `() => {x}; // ` is JSX content, not a function. I've added a negative test to help clarify.

Reply:
It's not surprising that it looks broken, because the test is supposed to look (to the parser) as much like an arrow function as possible without actually being one. I spent a lot of time in AST Explorer trying to find a minimally-differing pair of test cases (lines 206/213).

Reply:
(Also, there should arguably be some kind of scope for JSX text that we could use for a positive test here, but currently there isn't. Worth considering for a future PR.)